File: Assignment5/VB.NET/wordCounter.net/wordCounter.net/Form1.vb
Imports System.IO

Public Class Form1



    Private wDstr As SortedDictionary(Of String, Integer) = New SortedDictionary(Of String, Integer)()


    Private Sub Button1_Click_1(sender As Object, e As EventArgs) Handles Button1.Click
        Dim filePath = String.Empty
        OpenFileDialog1.InitialDirectory = "c:\"
        OpenFileDialog1.Filter = "txt files (*.txt)|*.txt|All files (*.*)|*.*"
        OpenFileDialog1.FilterIndex = 2
        OpenFileDialog1.RestoreDirectory = True
        OpenFileDialog1.Title = "Select Text File"

        If OpenFileDialog1.ShowDialog() = DialogResult.OK Then
            filePath = OpenFileDialog1.FileName

            Using sr As StreamReader = New StreamReader(filePath)

                While sr.Peek() >= 0
                    Dim line As String = sr.ReadLine()
                    Dim delimiters As Char() = New Char() {" "c, vbCr, vbLf}
                    Dim words As String() = line.Split(delimiters, StringSplitOptions.RemoveEmptyEntries)
                    Dim i As Integer = 0

                    For Each word As String In words

                        If Not wDstr.TryGetValue(word, i) Then
                            wDstr.Add(word, 1)
                        Else
                            wDstr.Remove(word)
                            i += 1
                            wDstr.Add(word, i)
                        End If
                    Next
                End While
            End Using

            Dim j As Integer = 0

            For Each w In wDstr
                Chart1.Series("Words").Points.AddXY(w.Key, w.Value)
            Next
        End If
    End Sub
End Class

File: Assignment5/C#/WordCounter/WordCounter/Form1.cs
using Microsoft.VisualBasic.FileIO;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Windows.Forms.DataVisualization.Charting;

namespace WordCounter
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
           
        }

       SortedDictionary<String, int> wDstr = new SortedDictionary<string, int>();
      
        
        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void button1_Click(object sender, EventArgs e)
        {
            var filePath = string.Empty;
            openFileDialog1.InitialDirectory = "c:\\";
            openFileDialog1.Filter = "txt files (*.txt)|*.txt|All files (*.*)|*.*";
            openFileDialog1.FilterIndex = 2;
            openFileDialog1.RestoreDirectory = true;
            openFileDialog1.Title = "Select Text File";

            if (openFileDialog1.ShowDialog() == DialogResult.OK)
            {
                //Get the path of specified file
                filePath = openFileDialog1.FileName;

                //Read the contents of the file into a stream
                using (StreamReader sr = new StreamReader(filePath))
                {
                    while (sr.Peek() >= 0)
                    {
                        string line = sr.ReadLine();
                        char[] delimiters = new char[] { ' ', '\r', '\n' };
                        string[] words = line.Split(delimiters, StringSplitOptions.RemoveEmptyEntries);
                        int i = 0;
                        foreach (string word in words)
                        {
                            if (!wDstr.TryGetValue(word, out i)) wDstr.Add(word, 1);
                            else { wDstr.Remove(word); i++; wDstr.Add(word, i); }
                            
                        }
                    }
                }
                int j = 0;
                foreach(var w in wDstr)
                {
                    chart1.Series["Words"].Points.AddXY(w.Key, w.Value);

                }


            }

        }

        private void chart1_Click(object sender, EventArgs e)
        {

        }
    }        
    }

